install.sh

```diff
@@ -277,16 +287,59 @@ elif [ $OS == 'centos' ] || [ $OS == 'redhat' ] || [ $OS == 'fedora'  ]; then
     yum install git -y
 fi
 
-if [ -f '/root/.my.cnf' ]
-then
-  MYSQL_ROOT_PASSWORD=$(awk -F "=" '/pass/ {print $2}' /root/.my.cnf)
-  echo " Password found at /root/.my.cnf, using $MYSQL_ROOT_PASSWORD"
-else
-  MYSQL_ROOT_PASSWORD=$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${2:-32};echo;)
-  echo " Generating new MySQL root password... $MYSQL_ROOT_PASSWORD"
+# Generate MySQL Password
+if [ "$TRAVIS" == "true" ]; then
+  echo "TRAVIS DETECTED! Setting empty 'root' user password."
+  MYSQL_ROOT_PASSWORD=''
   echo $MYSQL_ROOT_PASSWORD > /tmp/mysql_root_password
+  MAKEFILE_PATH="https://raw.githubusercontent.com/opendevshop/devshop/$DEVSHOP_VERSION/build-devmaster.make"
 fi
 
+# Determin if we create a db accoount or ask for it.
+# Starting from Debian 9 (MariaDB 10.1) there is passwordless login for root.
+# Therefore it needs special handling, Aegir does like to have an account with a password.
+# See https://www.drupal.org/node/2770819
+# TODO: Work with future versions
+if [ $OS == 'debian' ] && [ $VERSION == '9' ]; then
+  set -x
+  if [ "$TRAVIS" == "1" ]; then
+    # By default this policy file prevents starting daemons, which is useful during docker builds but not in this case.
+    rm /usr/sbin/policy-rc.d
+  fi
+  # Install MariaDB early to be able to create an account for Aegir.
+  apt-get install default-mysql-server --yes
+
+  # no password needed to login to the database ... MariaDB => 10.1 on Debian has this as default.
+  AEGIR_DB_USER="aegir_root"
```

Comment:
Make sure AEGIR_DB_USER is being set by default, as it's used down below in the Ansible variables.